### ISSUE TITLE
Implementation of TagRepository

### ISSUE THREAD

**Issue by webfirmframework** [webfirmframework/wff#31]
Implement a `TagRepository` class for `BrowserPage`, the concept [described in this page](https://github.com/webfirmframework/wff/wiki/8.-TagRepository).

The `TagRepository` must contain at least the following methods.

- `findTagById`
- `findTagsByAttribute`
- `findOneTagByAttribute`
- `findTagsByAttributeName`
- `findOneTagByAttributeName`

There must be a method to get the corresponding `TagRepository` from `BrowserPage` object, eg: `BrowserPage#getTagRepository`

**Comment by webfirmframework** [comment]
The methods mentioned in the description are implemented. 

The following methods are also implemented

- `upsert(final AbstractHtml tag, final String key,
            final WffBMArray bmArray)`
- `upsert(final AbstractHtml tag, final String key,
            final WffBMObject bmObject)`
- `delete(final AbstractHtml tag, final String key)`
Using the following methods, a JavaScript object / array may be set from server in the wffObjects property of a tag.

These changes will be available since _wffweb-2.1.8_

**Comment by webfirmframework** [comment]
Bug fixes and other performance improvements have been made to this class, it's available since _wffweb-2.1.9_.

**Comment by webfirmframework** [comment]
The following methods are available since _wffweb-2.1.11_.

```
findTagsByTagName(String tagName)
findOneTagByTagName(String tagName)
findOneTagAssignableToTag(Class)
findTagsAssignableToTag(Class)
findAttributesByTagName(String)
executeJs(String)
execute(BrowserPageAction)
```
